### PR TITLE
Fix/mouseenter event

### PR DESCRIPTION
**Pull request by stepancar**
Fix for https://github.com/facebook/react/issues/10109

### CODE REVIEW

**Comment by Andarist** [comment]
I would suspect that the 'sibling' event of `mouseleave` suffers from the same thing, this should be checked and fixed in the same PR I think (if indeed broken).

File: src/renderers/dom/shared/eventPlugins/EnterLeaveEventPlugin.js [hunk 1]
```diff
@@ -43,8 +43,9 @@ var EnterLeaveEventPlugin = {
     nativeEventTarget,
   ) {
     if (
-      topLevelType === 'topMouseOver' &&
-      (nativeEvent.relatedTarget || nativeEvent.fromElement)
+      (topLevelType === 'topMouseOver' &&
+        (nativeEvent.relatedTarget && !nativeEvent.relatedTarget.disabled)) ||
```
Comment (Andarist) [inline src/renderers/dom/shared/eventPlugins/EnterLeaveEventPlugin.js]: `!nativeEvent.relatedTarget.disabled` - is this needed? I think it wont even be emitted by the browser when `relatedTarget` is disabled

Comment (stepancar) [inline src/renderers/dom/shared/eventPlugins/EnterLeaveEventPlugin.js]: yes, its needed. Try to run it

**Comment by stepancar** [comment]
@Andarist https://github.com/facebook/react/pull/10149#issuecomment-314679615
mouseleave should not be called on disabled button

**Comment by gaearon** [comment]
I don't know if this is the right fix. Leaving to @aweary @jquense @nhunzaker to review.

**Comment by jquense** [comment]
I'm not sure this fixes the bug...it seems like it's fix an opposite problem. `mouseenter` on some sibling depends on `mouseout` from the `fromElement` element existing. Checking if the fromElement is disabled would assume it already exists, but the browser decided to fire the event despite the element being disabled (do any browsers do this?). Checking if the element is disabled would _prevent_ a mouseenter from firing, not cause it to

**Comment by gaearon** [comment]
Should we close this?

**Comment by jquense** [comment]
i think so...I'm trying to specifically identify what is wrong with the test.

**Comment by gaearon** [comment]
I'll leave this to you then :-P

**Comment by jquense** [comment]
Ok sorry I see what's happening. This change causes the plugin to take a path that is meant solely for the "entering an element from outside the browser window". and changes it to "there is an unpaired mouseover". It works for this one case, but I am not certain if it will lead to extra `mouseenters` being fired in other cases. It heavily depends on the browsers NOT firing the paired mouseout on the disabled button. If that changes or is not consistent across browsers we'll start getting double mouseenters

**Comment by gaearon** [comment]
I don't know. Unfortunately I don't think anyone on the team will have a lot of time this half to concentrate on the DOM issues, so if this isn't obviously the right fix, I think we won't take it.

**Comment by jquense** [comment]
ya fair enough. I generally thing the "right" fix is to use the native events 

Thanks for the PR @stepancar we want to try and address these mouseenter/leave issues a bit more holistically. I appreciate the contribution and if you'r interested please join in the conversation on #11972

**Pull request closed without merging**